Framework/HistogramData/test/HistogramIteratorTest.h

```diff
@@ -0,0 +1,412 @@
+
+    // Act
+    auto result =
+        std::equal(hist.begin(), hist.end(), expectedFrequencyVariances.begin(),
+                   [](const HistogramItem &item, const double &variance) {
+                     return item.frequencyVariance() == variance;
+                   });
+
+    // Assert
+    TSM_ASSERT("Frequency variances did not match", result);
+  }
+
+  void
+  test_iterate_over_histogram_frequency_variances_when_histogram_has_frequencies() {
+    // Arrange
+    Histogram hist(BinEdges{1, 2, 3, 5}, Frequencies{2, 3, 4},
+                   FrequencyVariances{3, 2, 1});
+    auto expectedFrequencyVariances = hist.frequencyVariances();
+
+    // Act
+    auto result =
+        std::equal(hist.begin(), hist.end(), expectedFrequencyVariances.begin(),
+                   [](const HistogramItem &item, const double &variance) {
+                     return item.frequencyVariance() == variance;
+                   });
+
+    // Assert
+    TSM_ASSERT("Frequency variances did not match", result);
+  }
+
+  void
+  test_iterate_over_histogram_frequency_std_when_histogram_has_frequencys() {
+    // Arrange
+    Histogram hist(BinEdges{1, 2, 3, 5}, Counts{2, 3, 4},
+                   CountVariances{3, 2, 1});
+    auto expectedFrequencyStd = hist.frequencyStandardDeviations();
+
+    // Act
+    auto result =
+        std::equal(hist.begin(), hist.end(), expectedFrequencyStd.begin(),
+                   [](const HistogramItem &item, const double &sigma) {
+                     return item.frequencyStandardDeviation() == sigma;
+                   });
+
+    // Assert
+    TSM_ASSERT("Frequency standard deviations did not match", result);
+  }
+
+  void
+  test_iterate_over_histogram_frequency_std_when_histogram_has_frequencies() {
+    // Arrange
+    Histogram hist(BinEdges{1, 2, 3, 5}, Frequencies{2, 3, 4},
+                   FrequencyVariances{3, 2, 1});
+    auto expectedFrequencyStd = hist.frequencyStandardDeviations();
+
+    // Act
+    auto result =
+        std::equal(hist.begin(), hist.end(), expectedFrequencyStd.begin(),
+                   [](const HistogramItem &item, const double &sigma) {
+                     return item.frequencyStandardDeviation() == sigma;
+                   });
+
+    // Assert
+    TSM_ASSERT("Frequency standard deviations did not match", result);
+  }
+};
+
+class HistogramIteratorTestPerformance : public CxxTest::TestSuite {
+public:
+  static HistogramIteratorTestPerformance *createSuite() {
+    return new HistogramIteratorTestPerformance;
+  }
+  static void destroySuite(HistogramIteratorTestPerformance *suite) {
+    delete suite;
+  }
+
+  HistogramIteratorTestPerformance() {
+    BinEdges edges(histSize, LinearGenerator(0, 2));
+    Counts counts(histSize - 1, LinearGenerator(0, 2));
+    for (size_t i = 0; i < nHists; i++)
+      hists.push_back(Histogram(edges, counts));
+  }
+
+  void test_iterate_and_access_each_item() {
+    double total = 0;
+    for (auto &hist : hists)
```

Comment:
I think I would reuse the same histogram. Using different histograms makes mainly sense if you are interested in the memory bandwidth of your computer, I think.